### DISCUSSION TITLE
Is this hard-coded "DAGS_FOLDER" string a bug in TaskInstance regarding DAG path?

### DISCUSSION THREAD

**Discussion by LipuFei** [apache/airflow#37544] [General] [answered]
I was experimenting with the AWS ECS Executor, but the tasks always fail. Then I saw that the ECS task commands have `DAGS_FOLDER/...` as the dag path. This `DAGS_FOLDER` seems to be a "hard-coded mistake".

A quick search took me to this line: https://github.com/apache/airflow/blob/f2ea8a3e1753012bfe0d529c9c8be66cf55ca28f/airflow/models/taskinstance.py#L1521-L1530

Line 1530 has a hard-coded `"DAGS_FOLDER"` string.

I don't know the history or intention here, so it's difficult for me to say whether it's a bug or not. Does anyone know?

**UPDATE:** I feel that it should be this instead?
```python
if path:
    if not path.is_absolute():
        path = conf.get_mandatory_value("core", "DAGS_FOLDER") / path
```

Thanks in advance.

AWS ECS Executor doc link: https://airflow.apache.org/docs/apache-airflow-providers-amazon/8.17.0/executors/ecs-executor.html

**Comment by Taragolis** [comment, marked as answer]
Nope this method generate command-line arguments which will provided into the airflow worker later and replaced by actual folder 

https://github.com/apache/airflow/blob/2f07c6b7173bf6909db6c25e812bfa95018d8182/airflow/utils/cli.py#L175-L182

**Comment by LipuFei** [comment]
Thanks for the answer 👍 @Taragolis
